tests/backup/backup_test.go

```diff
@@ -177,45 +173,32 @@ var _ = Describe("{BasicBackupCreation}", func() {
 		}
 	})
 	It("Basic Backup Creation", func() {
-		Step("Validate applications and get their labels", func() {
+		Step("Validate applications", func() {
 			ValidateApplications(contexts)
-			log.Infof("Create list of pod selector for the apps deployed")
-			for _, ctx := range appContexts {
-				for _, specObj := range ctx.App.SpecList {
-					if obj, ok := specObj.(*appsapi.Deployment); ok {
-						if Contains(appList, obj.Name) {
-							ps[obj.Name] = obj.Spec.Template.Labels
-						}
-					} else if obj, ok := specObj.(*appsapi.StatefulSet); ok {
-						if Contains(appList, obj.Name) {
-							ps[obj.Name] = obj.Spec.Template.Labels
-						}
-					}
-				}
-			}
 		})
-
 		Step("Creating rules for backup", func() {
-			var err error
 			log.InfoD("Creating pre rule for deployed apps")
-			preRuleStatus, preRuleUid, err = Inst().Backup.CreateRuleForBackup(PreRuleName, "default", appList,
-				"pre", ps)
-			log.FailOnError(err, "Creating pre rule for deployed apps failed")
-			dash.VerifyFatal(preRuleStatus, true, "Verifying pre rule for backup")
+			for i := 0; i < len(appList); i++ {
+				preRuleStatus, err, ruleName := Inst().Backup.CreateRuleForBackup(appList[i], orgID, "pre")
+				preRuleNameList = append(preRuleNameList, ruleName)
+				log.FailOnError(err, "Creating pre rule for deployed apps failed")
```

Comment:
handle the error first and then use ruleName

Reply:
done